openssl/src/asn1/mod.rs

```diff
@@ -18,6 +22,13 @@ impl Asn1Time {
         }
     }
 
+    pub fn unowned(handle: *mut ffi::ASN1_TIME) -> Asn1Time {
```

Comment:
If we're going to add something like this we'll definitely need to add a lifetime paramter to `Asn1Time` to enforce that it can't be used after the structure that does own the `ASN1_TIME` has been destroyed. Take a look at the `x509` module - I overhauled that yesterday to have a `X509Ref<'a>` for unowned use cases and a `X509` type for owned use cases which derefs to `X509Ref<'static>`.